### PR TITLE
change and add some logs in generator dispatch modification

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java [hunk 1]
```diff
@@ -148,53 +148,70 @@ private static Double getGeneratorMarginalCost(Generator generator) {
...
+                        Map.of(GENERATOR, generator.getId()), TypedValue.TRACE_SEVERITY);
             }
             return marginalCost != null;
         }).collect(Collectors.toList());
 
+        int nbNoCost = generators.size() - generatorsWithMarginalCost.size();
+        if (nbNoCost > 0) {
+            report(reporter, reporterSuffixKey, "NbGeneratorsWithNoCost", "${nbNoCost} generator${isPlural} been discarded from generation dispatch because of missing marginal cost. Their active power set point has been set to 0",
+                    Map.of("nbNoCost", nbNoCost,
+                            "isPlural", nbNoCost > 1 ? "s have" : " has"),
```
Comment (Walid-Sahnoun) [inline src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java:171]: "isPlural" is used for 5 times in this file, use a constant will be better ?

File: src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java [hunk 2]
```diff
@@ -379,13 +414,42 @@ private double reduceGeneratorMaxPValue(Generator generator,
...
+        disconnectedGenerators.stream()
+                .filter(g -> g.getTerminal().getBusView() != null && g.getTerminal().getBusView().getConnectableBus() != null &&
+                        g.getTerminal().getBusView().getConnectableBus().getSynchronousComponent().getNum() == componentNum)
+                .forEach(g -> {
+                    report(reporter, Integer.toString(componentNum), "DisconnectedGenerator", "Generator ${generator} has been discarded from generation dispatch because it is disconnected. Its active power set point remains unchanged",
+                            Map.of(GENERATOR, g.getId()), TypedValue.TRACE_SEVERITY);
+                    disconnectedGeneratorCounter.getAndIncrement();
+                });
+        if (disconnectedGeneratorCounter.get() > 0) {
+            report(reporter, Integer.toString(componentNum), "TotalDisconnectedGenerator", "${nbDisconnectedGenerator} generator${isPlural} been discarded from generation dispatch because their are disconnected. Their active power set point remains unchanged",
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java:428]: Add the synthetic report before each individual disconnected generator report ?

File: src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java [hunk 3]
```diff
@@ -148,53 +148,70 @@ private static Double getGeneratorMarginalCost(Generator generator) {
...
-                report(reporter, Integer.toString(component.getNum()), "MissingMarginalCostForGenerator", "The generator ${generator} does not have a marginal cost",
-                    Map.of(GENERATOR, generator.getId()), TypedValue.WARN_SEVERITY);
+                report(reporter, reporterSuffixKey, "MissingMarginalCostForGenerator", "The generator ${generator} does not have a marginal cost",
+                        Map.of(GENERATOR, generator.getId()), TypedValue.TRACE_SEVERITY);
             }
             return marginalCost != null;
         }).collect(Collectors.toList());
 
+        int nbNoCost = generators.size() - generatorsWithMarginalCost.size();
+        if (nbNoCost > 0) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java:168]: Add this synthetic report before each individual generator without marginal cost report ?

File: src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java [hunk 4]
```diff
@@ -268,8 +286,25 @@ public void onUpdate(Identifiable identifiable, String attribute, String variant
...
+        public void endReport(List<Generator> adjustableGenerators) {
+            report(reporter, suffixKey, "TotalGeneratorSetTargetP", "The active power set points of ${nbUpdatedGenerator} generator${isPlural} have been updated as a result of generation dispatch",
+                    Map.of("nbUpdatedGenerator", updatedGenerators.size(), "isPlural", updatedGenerators.size() > 1 ? "s" : ""), TypedValue.INFO_SEVERITY);
+            // what are unchanged generators ?
+            adjustableGenerators.stream()
+                .filter(g -> !updatedGenerators.contains(g.getId()))
+                .forEach(g -> report(reporter, suffixKey, "GeneratorUnchangedTargetP", "Generator ${generator} has not been selected by the merit order algorithm. Its active power set point has been set to 0",
+                        Map.of(GENERATOR, g.getId()), TypedValue.TRACE_SEVERITY));
+            int nbUnchangedGenerators = adjustableGenerators.size() - updatedGenerators.size();
+            if (nbUnchangedGenerators > 0) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java:304]: Add this synthetic report before each individual unchanged generator report ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java (hunk 4 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java:299]: also made for update generators (synthetic before details)

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**